runtime/amd64.S

```diff
@@ -1353,15 +1353,21 @@ CFI_STARTPROC
             16 /* exn */ +                            \
             8 /* gc_regs slot (unused) */ +           \
             8 /* C_STACK_SP for DWARF (unused) */ +   \
-            8 * Stack_padding_word +                  \
+            16 * Stack_padding_word +                  \
             Handler_parent, DW_OP_deref,              \
           DW_OP_plus_uconst, Stack_sp, DW_OP_deref,   \
           DW_OP_plus_uconst, RETADDR_ENTRY_SIZE
 #endif
         movq    %rdi, %rax /* first argument */
+        CHECK_STACK_ALIGNMENT;
         callq   *(%rbx) /* closure in %rbx (second argument) */
 LBL(frame_runstack):
         leaq    32(%rsp), %r11 /* SP with exn handler popped */
+#if Stack_padding_word
+    /* Pop the additional alignment padding word *and* the stack padding word,
+       so that $r11 points to the handler struct. */
+        addq    $16, %r11
```

Comment:
It took me a long time to grasp what was going on here. A diagram would really help!